cypress/e2e/custom/ecospheres/topics/elements/deeplink.cy.ts

```diff
@@ -0,0 +1,186 @@
+
+describe('Topic Elements - Deep Linking', () => {
+  let testTopic: Topic
+  let testFactors: Factor[]
+
+  beforeEach(() => {
+    ;({ testTopic, testFactors } = setupTopicWithExistingFactors())
+  })
+
+  describe('Navigating to a specific factor via URL hash', () => {
+    it('should navigate to a factor in a group when hash is present in URL', () => {
+      const targetFactor = testFactors[0]
+      const factorId = targetFactor.id
+
+      // Visit the topic page with a hash pointing to the factor
+      cy.visit(`/bouquets/${testTopic.id}#factor-${factorId}`)
+
+      cy.wait('@getElementsDataset')
+
+      // Verify the factor is visible (group should auto-expand)
+      cy.contains(targetFactor.title).should('be.visible')
+
+      // Verify the factor element has the correct ID
+      cy.get(`#factor-${factorId}`).should('exist')
+
+      // Verify the factor is scrolled into viewport
+      // Wait for smooth scroll animation to complete
+      cy.get(`#factor-${factorId}`).isInViewport({ wait: VIEWPORT_WAIT })
+
+      // Verify we're on the "Données" tab
+      cy.get('[role="tablist"]')
+        .find('[role="tab"][aria-selected="true"]')
+        .should('contain', 'Données')
+    })
+
+    it('should highlight the target factor temporarily', () => {
+      const targetFactor = testFactors[0]
+      const factorId = targetFactor.id
+
+      cy.visit(`/bouquets/${testTopic.id}#factor-${factorId}`)
+      cy.wait('@getElementsDataset')
+
+      // Wait for smooth scroll animation to complete
+      cy.get(`#factor-${factorId}`).isInViewport({ wait: VIEWPORT_WAIT })
+
+      // Verify the factor element gets highlighted
+      cy.get(`#factor-${factorId}`).should(($el) => {
+        const outline = $el.css('outline')
+        // Check that outline is applied (it should be non-empty)
+        expect(outline).to.not.include('0px')
+      })
+
+      // Wait for highlight to disappear (2 seconds + buffer)
+      cy.wait(2500)
+
+      // Verify the highlight is removed
+      cy.get(`#factor-${factorId}`).should(($el) => {
+        const outline = $el.css('outline')
+        // Outline should be removed or set to none/initial
+        expect(outline).to.include('0px')
+      })
+    })
+
+    it('should work with multiple factors in the same group', () => {
+      const firstFactor = testFactors[0]
+      const secondFactor = testFactors[1]
+
+      // Navigate to first factor
+      cy.visit(`/bouquets/${testTopic.id}#factor-${firstFactor.id}`)
+      cy.wait('@getElementsDataset')
+
+      cy.get(`#factor-${firstFactor.id}`).isInViewport({ wait: VIEWPORT_WAIT })
+      cy.contains(firstFactor.title).should('be.visible')
+
+      // Now navigate to second factor by updating the hash
+      cy.visit(`/bouquets/${testTopic.id}#factor-${secondFactor.id}`)
+
+      cy.get(`#factor-${secondFactor.id}`).isInViewport({ wait: VIEWPORT_WAIT })
+      cy.contains(secondFactor.title).should('be.visible')
```

Comment:
This passes even if I comment out the 2nd `cy.visit`. Related to my 1st comment above.

Reply:
Removed the test, not related to a use case.